plugins/inputs/intel_powerstat/services/rapl.go

```diff
@@ -0,0 +1,274 @@
+	energyUjPath := fmt.Sprintf(EnergyUjPartialPath, raplPath)
+	energyUjFile, err := os.Open(energyUjPath)
+	if err != nil {
+		return fmt.Errorf("error opening energy_uj file on path %s, err: %v", energyUjPath, err)
+	}
+	defer energyUjFile.Close()
+
+	dramPath := fmt.Sprintf(IntelRaplDramPartialPath, IntelRaplPath, socketID, r.dramFolders[socketID])
+	dramEnergyPath := fmt.Sprintf(EnergyUjPartialPath, dramPath)
+	dramEnergyFile, err := os.Open(dramEnergyPath)
+	if err != nil {
+		return fmt.Errorf("error opening dram energy_uj file on path %s, err: %v", dramEnergyPath, err)
+	}
+	defer dramEnergyFile.Close()
+
+	return r.calculateData(socketID, energyUjFile, dramEnergyFile)
+}
+
+// GetConstraintMaxPower retrieves ConstraintMaxPower from constraint_0_max_power_uw file.
+func (r *RaplServiceImpl) GetConstraintMaxPower(socketID string) (float64, error) {
+	raplPath := fmt.Sprintf(IntelRaplSocketPartialPath, IntelRaplPath, socketID)
+	maxPowerPath := fmt.Sprintf(MaxPowerUwPartialPath, raplPath)
+	maxPowerFile, err := os.Open(maxPowerPath)
+	if err != nil {
+		return 0, fmt.Errorf("error opening constraint_0_max_power_uw file on path %s, err: %v", maxPowerPath, err)
+	}
+	defer maxPowerFile.Close()
+	maxPower, _, err := r.fs.ReadFileToFloat64(maxPowerFile)
+
+	return maxPower, err
+}
+
+// GetDramMaxEnergyRangeUj retrieves dram energy range value from max_energy_range_uj file.
+func (r *RaplServiceImpl) GetDramMaxEnergyRangeUj(socketID string) (float64, error) {
+	dramPath := fmt.Sprintf(IntelRaplDramPartialPath, IntelRaplPath, socketID, r.dramFolders[socketID])
+	dramEnergyPath := fmt.Sprintf(MaxRangeUjPartialPath, dramPath)
+	maxDramEnergyFile, err := os.Open(dramEnergyPath)
+	if err != nil {
+		return 0, fmt.Errorf("error opening max_energy_range_uj file on path %s, err: %v", dramEnergyPath, err)
+	}
+	defer maxDramEnergyFile.Close()
+	maxDramEnergy, _, err := r.fs.ReadFileToFloat64(maxDramEnergyFile)
+
+	return maxDramEnergy, err
+}
+
+// GetMaxEnergyRangeUj retrieves energy range value from max_energy_range_uj file.
+func (r *RaplServiceImpl) GetMaxEnergyRangeUj(socketID string) (float64, error) {
+	raplPath := fmt.Sprintf(IntelRaplSocketPartialPath, IntelRaplPath, socketID)
+	maxEnergyPath := fmt.Sprintf(MaxRangeUjPartialPath, raplPath)
+	maxEnergyFile, err := os.Open(maxEnergyPath)
+	if err != nil {
+		return 0, fmt.Errorf("error opening max_energy_range_uj file on path %s, err: %v", maxEnergyPath, err)
+	}
+	defer maxEnergyFile.Close()
+	maxEnergy, _, err := r.fs.ReadFileToFloat64(maxEnergyFile)
+
+	return maxEnergy, err
+}
+
+// getDramFolders returns Dram folder names from RAPL.
+func (r *RaplServiceImpl) getDramFolders() map[string]string {
+	return r.dramFolders
+}
+
+func (r *RaplServiceImpl) prepareData() {
+	intelRaplPrefix := "intel-rapl:"
+	intelRapl := fmt.Sprintf("%s%s", intelRaplPrefix, "[0-9]*")
+	raplPaths, err := r.fs.GetStringsMatchingPatternOnPath(fmt.Sprintf("%s/%s", IntelRaplPath, intelRapl))
```

Comment:
There should never be an error as the underlying `filepath.Glob()` only returns an error on malformed patterns. As you specify the pattern this should never happen. So return an error if there is a problem as probably more things are broken then.

Reply:
I understand, this is almost impossible to happen that this pattern is not well-formatted but for security reasons and for the sake of handling every error we would like to leave it as it is.